src/main/scala/io/citrine/lolo/bags/BaggedResult.scala

```diff
@@ -434,11 +437,159 @@ case class MultiTaskBaggedResult(baggedPredictions: Seq[BaggedResult[Any]], real
     }.transpose)
   }
 
-  override def getUncertaintyCorrelation(i: Int, j: Int): Option[Seq[Double]] = {
+  /** For now, call the trivial implementation. */
+  override def getUncertaintyCorrelation(i: Int, j: Int): Option[Seq[Double]] = getUncertaintyCorrelationTrivial(i, j)
+
+  /**
+    * Calculate the uncertainty correlation using one of a variety of methods.
+    * This method ensures that the two labels are distinct and both real-valued, and then dispatches the calculation
+    * to the specified method.
+    *
+    * @param i      index of the first label
+    * @param j      index of the second label
+    * @param method method by which to calculate correlation
+    * @return       optional sequence of correlation coefficients between specified labels for each prediction
+    */
+  def getUncertaintyCorrelationBuffet(i: Int, j: Int, method: CorrelationMethod): Option[Seq[Double]] = {
     (realLabels(i), realLabels(j)) match {
       case (true, true) if i == j => Some(Seq.fill(numPredictions)(1.0))
-      case (true, true) if i != j => Some(Seq.fill(numPredictions)(0.0))
-      case _: Any                 => None
+      case (true, true) => method match {
+        case Trivial => getUncertaintyCorrelationTrivial(i, j)
+        case FromTraining => getUncertaintyCorrelationTraining(i, j)
+        case Bootstrap => getUncertaintyCorrelationBootstrap(i, j)
+        case Jackknife => getUncertaintyCorrelationJackknife(i, j)
+        case JackknifeExplicit => getUncertaintyCorrelationJackknifeExplicit(i, j)
+      }
+      case _: Any => None
+    }
+  }
+
+  /** Uncertainty correlation is always 0.0 */
+  private def getUncertaintyCorrelationTrivial(i: Int, j: Int): Option[Seq[Double]] = {
```

Comment:
Should this be 1 on the diagonal?

Reply:
That's taken care of by the first case of `getUncertaintyCorrelationBuffet`